x-pack/plugins/infra/public/hooks/use_log_view.ts

```diff
@@ -67,6 +67,7 @@ export const useLogView = ({
     () => ({
       fields: resolvedLogView?.fields ?? [],
       title: resolvedLogView?.indices ?? 'unknown',
+      type: 'title',
```

Comment:
I think you might be on the right track here, but it doesn't quite seem to work correctly. When I look at the completions they are actually fields from the metrics indices and not the logs. By inspecting the network requests I could see that it searches for an index pattern `"undefined"`:

![image](https://user-images.githubusercontent.com/973741/197579551-e047b25f-8db5-4b88-b120-3e5b4956cdba.png)

Ultimately that leads to it using the "default" index pattern:

https://github.com/elastic/kibana/blob/b921a49b3b2b68ac7d14181962b2c7a7b70f703d/src/plugins/unified_search/public/query_string_input/fetch_index_patterns.ts#L50

Tracing back the origin of that `undefined`, it seems for `type === 'title'` it reads `.value`, which we don't pass.

I hope we'll manage to get by with a small fix, but nothing seems obvious to me. The fact that we're casting our derived data view `as DataView[]` when passing it to the search bar means we're loosing all type safety. :thinking:

Reply:
@weltenwort 

(╯°□°）╯︵ ┻━┻

It looks like there's a way to fix this easily for the query (with a gross TS cast), but as soon as we bring in filters I think there will be problems.

[In the top level `SearchBar` the `indexPatterns` are typed as an array of `DataView`s](https://github.com/elastic/kibana/blob/main/src/plugins/unified_search/public/search_bar/search_bar.tsx#L45). 

However, at the next level down in the hierarchy the [query components are actually setup to take an array of DataViews **or** strings](https://github.com/elastic/kibana/blob/main/src/plugins/unified_search/public/query_string_input/query_bar_top_row.tsx#L64). 

I tried out just passing in an array of strings and this did actually work:

![Screenshot 2022-10-25 at 11 36 54](https://user-images.githubusercontent.com/471693/197752022-a45f0642-3b36-47db-a311-a809466c86ee.png)

Looking back at the `SearchBar` though, [the semantics of it being a `DataView` do inform some decisions](https://github.com/elastic/kibana/blob/main/src/plugins/unified_search/public/search_bar/search_bar.tsx#L250). Again, we don't need this *yet* but we do want it soon-ish.

[And we would have problems by the looks of it once filters are turned on in the issues coming up](https://github.com/elastic/kibana/blob/main/src/plugins/unified_search/public/filter_bar/filter_bar.tsx#L22).

Having a real DataView is for sure the nicest way to move forward but I'm not sure how to deal with the change in semantics around indices not existing, that's a very, very big change in behaviour.

I could just wrap the `dataViewsService.create()` call in in a try / catch, and set `showErrors` to `false`. But I'm not sure what the fallback should be in the `catch()`, the only real option is to set it to `undefined`, and then I guess the bar just loses all value.

Reply:
@weltenwort 

The mixture of https://github.com/elastic/kibana/pull/143222/commits/6256bc828092d8487cc13be2663e705e64819684 and https://github.com/elastic/kibana/pull/143222/commits/3595f0fbb3c358e72a05eecb494efaf6d1af8196 contains the changes we discussed earlier, and AFAICS they work as intended.

Overview:

When resolving a legacy reference `dataViewsService.create()` will be called (with `allowNoIndex: true` which was under the `spec`). The `id` is set in a deterministic manner using the `LogView` ID and a prefix that should make it easy to track to us. This is then set as the `dataViewReference` on the `ResolvedLogView`.

The manual `fields` resolution is removed in favour of passing through the fields from the `DataView`.

`displayErrors` is set to `false` when calling `create()` in favour of using our specific `ResolveLogViewError`, so that this is handled using our proper error component. E.g.

![Screenshot 2022-10-25 at 17 00 23](https://user-images.githubusercontent.com/471693/197824203-b6e41f4e-02f9-4e12-bdaa-e6c348e1f54b.png)

(The error message there is from me doing a manual `throw new Error()` inside of `.create()`.)

Network requests all look sound, and the suggestions themselves:

![Screenshot 2022-10-25 at 16 36 25](https://user-images.githubusercontent.com/471693/197824372-d65d1ae2-237e-4d1d-8e89-c065dc0a8907.png)

And I also threw some console logs inside of `_pattern_cache.ts` to make sure we're only hitting the cache with one key when navigating between several solutions (but in the same session).

If we're happy with the approach I'll amend the tests.